### PR TITLE
chore(openapi): upgrade from ir v23 -> v53

### PR DESCRIPTION
**Pull request by dsinghvi**
This PR upgrades the OpenAPI generator to IR v53 so that we can start improving the OpenAPI generator.

### CODE REVIEW

File: seed/openapi/streaming/openapi.yml [hunk 1]
```diff
@@ -9,7 +9,9 @@ paths:
       tags:
         - Dummy
       parameters: []
-      responses: {}
```
Comment (dsinghvi) [inline seed/openapi/streaming/openapi.yml:12]: responses: {} is invalid

File: generators/openapi/versions.yml [hunk 2]
```diff
@@ -1,3 +1,13 @@
+- version: 0.1.0
+  createdAt: '2024-03-24'
```
Comment (armandobelardo) [inline generators/openapi/versions.yml:2]: Nit: could we remove or update this

File: generators/openapi/versions.yml [hunk 3]
```diff
@@ -1,3 +1,13 @@
...
+  changelogEntry:
+    - type: chore
```
Comment (armandobelardo) [inline generators/openapi/versions.yml:4]: Thoughts on `internal` as the type here?

File: seed/openapi/query-parameters/openapi.yml [hunk 4]
```diff
@@ -40,16 +40,16 @@ paths:
             format: byte
         - name: user
           in: query
-          required: false
+          required: true
           schema:
-            type: string
-            nullable: true
+            $ref: '#/components/schemas/User'
```
Comment (dsinghvi) [inline seed/openapi/query-parameters/openapi.yml:45]: we now support objects as query params

File: seed/openapi/query-parameters/openapi.yml [hunk 5]
```diff
@@ -40,16 +40,16 @@ paths:
             format: byte
         - name: user
           in: query
-          required: false
+          required: true
```
Comment (dsinghvi) [inline seed/openapi/query-parameters/openapi.yml:43]: we now respect required query params

File: seed/openapi/plain-text/openapi.yml [hunk 6]
```diff
@@ -10,11 +10,8 @@ paths:
         - Service
       parameters: []
       responses:
-        '200':
+        '204':
```
Comment (dsinghvi) [inline seed/openapi/plain-text/openapi.yml:13]: we dont handle text responses so this is a followup

File: generators/openapi/src/security.ts
Comment (armandobelardo) [inline generators/openapi/src/security.ts:1]: Nit: do we want to add the whole `flows` block with oauth into the spec?

File: generators/openapi/src/converters/typeConverter.ts [hunk 7]
```diff
@@ -38,7 +39,7 @@ export function convertType(typeDeclaration: TypeDeclaration, ir: IntermediateRe
             return convertEnum({ enumTypeDeclaration, docs });
         },
         object: (objectTypeDeclaration) => {
-            const exampleType: ExampleType | undefined = typeDeclaration.examples[0];
+            const exampleType: ExampleType | undefined = typeDeclaration.userProvidedExamples[0];
```
Comment (armandobelardo) [inline generators/openapi/src/converters/typeConverter.ts:42]: Should we default to generated if no user provided?

File: generators/openapi/src/converters/typeConverter.ts [hunk 8]
```diff
@@ -312,7 +342,7 @@ function convertContainerType(containerType: ContainerType): OpenApiComponentSch
         map: (mapType) => {
             if (
                 mapType.keyType.type === "primitive" &&
-                mapType.keyType.primitive === "STRING" &&
+                mapType.keyType.primitive.v1 === "STRING" &&
```
Comment (armandobelardo) [inline generators/openapi/src/converters/typeConverter.ts:345]: Do you need to check the v2 type as well?

File: generators/openapi/src/converters/servicesConverter.ts [hunk 9]
```diff
@@ -375,11 +391,15 @@ function convertResponse({
...
 
         responseByStatusCode["200"] = {
-            description: httpResponse.docs ?? "",
+            description: httpResponse.body.value.docs ?? "",
             content: {
                 "application/json": convertedResponse
             }
         };
+    } else {
+        responseByStatusCode["204"] = {
```
Comment (armandobelardo) [inline generators/openapi/src/converters/servicesConverter.ts:400]: What's this about? (e.g. why is 204 the fallback)

**Review by armandobelardo — APPROVED** [review]

**Pull request merged**